mlir/test/Dialect/ArmSME/enable-arm-streaming.mlir

```diff
@@ -9,3 +9,11 @@
 // CHECK-ENABLE-ZA-LABEL: @arm_streaming
 // CHECK-ENABLE-ZA-SAME: attributes {arm_streaming, arm_za}
 func.func @arm_streaming() { return }
+
+// CHECK-LABEL: @not_arm_streaming
+// CHECK-SAME: attributes {enable_arm_streaming_ignore}
+// CHECK-LOCALLY-LABEL: @not_arm_streaming
+// CHECK-LOCALLY-SAME: attributes {enable_arm_streaming_ignore}
+// CHECK-ENABLE-ZA-LABEL: @not_arm_streaming
+// CHECK-ENABLE-ZA-SAME: attributes {enable_arm_streaming_ignore}
```

Comment:
I was after something like this - I think that `CHECK-NOT` is more descriptive in this context (i.e., we want to make sure that in presence of `enable_arm_streaming_ignore` we won't see these streaming SVE attributes)

```suggestion
// CHECK-LABEL: @not_arm_streaming
// CHECK-NOT: arm_streaming
// CHECK-LOCALLY-LABEL: @not_arm_streaming
// CHECK-LOCALLY-NOT: arm_locally_streaming
// CHECK-ENABLE-ZA-LABEL: @not_arm_streaming
// CHECK-ENABLE-ZA-NOT: arm_za
```

Reply:
I've added CHECK-NOTs between the matches now (still with the CHECK-SAMEs too). I'm a little wary of just using CHECK-NOT alone (since if it's matching checking the wrong line it can falsely pass too).

Reply:
The way this is written now, I think that you could have the following output and the test would pass (which is not what we want):

```
func.func @not_arm_streaming() attributes {enable_arm_streaming_ignore, arm_locally_streaming} { return }
```

So you'd need this to be 100% sure:
```
// CHECK-LABEL: @not_arm_streaming
// CHECK-NOT: arm_streaming
// CHECK-LOCALLY-LABEL: @not_arm_streaming
// CHECK-NOT: arm_streaming
```

This should be sufficient though:
```
// CHECK-LABEL: @not_arm_streaming
// CHECK-NOT: arm_streaming
```
This way, you match the label (which is unique) and the make sure that there's no `arm_streaming` attribute starting from the label all the way to EOF.

Reply:
You have to do:
```
// CHECK-LABEL: @not_arm_streaming
// CHECK-NOT: arm_streaming
// CHECK-LOCALLY-LABEL: @not_arm_streaming
// CHECK-NOT: arm_streaming
```
Because `arm_streaming` is a substring of `enable_arm_streaming_ignore`. There's probably some syntax for whole word matches, but I think that just using ` // CHECK-SAME:` is a little simpler and easier to understand here.

Reply:
> There's probably some syntax for whole word matches

There's [-match-full-lines](https://www.llvm.org/docs/CommandGuide/FileCheck.html#cmdoption-FileCheck-match-full-lines)